tests/api/test_isend_irecv.py

```diff
@@ -0,0 +1,312 @@
+# pylint: disable=missing-function-docstring,missing-class-docstring,missing-module-docstring,too-many-arguments
+import time
+
+import numpy as np
+import pytest
+from mpi4py.MPI import COMM_WORLD
+
+import numba_mpi as mpi
+from tests.common import data_types
+from tests.utils import get_random_array
+
+TEST_WAIT_FULL_IN_SECONDS = 0.3
+TEST_WAIT_INCREMENT_IN_SECONDS = 0.1
+
+
+@pytest.mark.parametrize(
+    "isnd, ircv, wait",
+    (
+        (mpi.isend, mpi.irecv, mpi.wait),
+        (mpi.isend.py_func, mpi.irecv.py_func, mpi.wait.py_func),
```

Comment:
using `py_func` here ensures that we have `jit` decorators for all tested functions, but we do not ensure that `njit`
to this end, in other test files, we started introducing test-file-defined njitted wrappers like here: https://github.com/numba-mpi/numba-mpi/blob/6fc280addaf4c82f275a89f191bf06ba9a8a8807/tests/api/test_barrier.py#L10

Reply:
done for all relevant instances - especially required for some functions that started to use overloads